lib/cloudera-manager/service.rb

```diff
@@ -1,4 +1,64 @@
 module ClouderaManager
   class Service < BaseResource
+    class RestartFailedException < BaseException; end
+
+    belongs_to :cluster
+    has_many :roles
+    primary_key :name
+    collection_path "clusters/:cluster_id/services"
+    after_find { |s| s.cluster_id = s.clusterRef[:clusterName] }
+
+    BULK_RESTART_SAFE_ROLE_TYPES = [
+      "IMPALAD",
+      "DATANODE",
+      "NODEMANAGER"
+    ]
+
+    RESTART_ORDER = {
+      "YARN" => ["JOBHISTORY", "NODEMANAGER", "RESOURCEMANAGER"],
+      "HDFS" => ["SECONDARYNAMENODE", "HTTPFS", "FAILOVERCONTROLLER", "NFSGATEWAY", "DATANODE", "NAMENODE", "JOURNALNODE"],
```

Comment:
I suppose so! Do we want to decide that we always restart the standby first to get it the new config, and then fail over to it, and then restart the other one? I guess that is better than potentially restarting the active one to only have it  potentially not come back up and interrupt a bunch of work that was underway. Restarting the standby and having it not come back up for whatever reason is definitely preferable.